src/test/java/com/salesforce/storm/spout/dynamic/VirtualSpoutTest.java

```diff
@@ -1608,6 +1608,53 @@ public void testGetCurrentState() {
         assertEquals("Should be our expected instance", expectedConsumerState, result);
     }
 
+    /**
+     * Test that ending state can be set after the {@link VirtualSpout} is opened.
+     */
+    @Test
+    public void testSetEndingState() {
+        // Create inputs
+        final Map topologyConfig = getDefaultConfig();
+        final TopologyContext mockTopologyContext = new MockTopologyContext();
+        final RetryManager mockRetryManager = mock(RetryManager.class);
+
+        // Create a mock SidelineConsumer
+        final Consumer mockConsumer = mock(Consumer.class);
+
+        // Create factory manager
+        final FactoryManager mockFactoryManager = createMockFactoryManager(null, mockRetryManager, null, null);
+        when(mockFactoryManager.createNewConsumerInstance()).thenReturn(mockConsumer);
+
+        // Create spout & open
+        final VirtualSpout virtualSpout = new VirtualSpout(
+            new DefaultVirtualSpoutIdentifier("MyConsumerId"),
+            topologyConfig,
+            mockTopologyContext,
+            mockFactoryManager,
+            new LogRecorder(),
+            ConsumerState.builder().build(),
+            // Ending state here is explicitly set to null
+            null
+        );
+        virtualSpout.open();
+
+        assertNull("Ending state is not null", virtualSpout.getEndingState());
+
+        final ConsumerState endingState = ConsumerState
+            .builder()
+            .withPartition("Test", 0, 100L)
+            .build()
+        ;
```

Comment:
where to have a dangling ; no?

Reply:
I almost always dangle my semi-colons on multi-liners like this, makes it easy to comment a line out or reorder them. I can move it to the last line it if you like, just let me know.